elm/web/rhub.py

```diff
@@ -0,0 +1,601 @@
+"""
+Code to build Corpus from the researcher hub.
+"""
+import os
+import os.path
+import logging
+from urllib.request import urlopen
+import requests
+import pandas as pd
+from bs4 import BeautifulSoup
+from rex import init_logger
+
+logger = logging.getLogger(__name__)
+init_logger(__name__, log_level='DEBUG')
+init_logger('elm', log_level='INFO')
+
+
+class ResearchOutputs():
+    """Class to handle publications portion of the NREL researcher hub."""
+    BASE_URL = "https://research-hub.nrel.gov/en/publications/?page=0"
+
+    def __init__(self, url, n_pages=1, txt_dir='./ew_txt'):
```

Comment:
Please include a docstring for dunder methods. It's not required by pylint because dunder methods are "hidden" but they're often the most important methods. For init, you don't need the one line description, just:
```
"""
Parameters
----------
url : str
    description
...
"""